src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c

```diff
@@ -36,16 +37,132 @@ static InitChainEntry D_80ABD750[] = {
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+Vec3f D_80ABD760[] = { 0.0f, 0.0f, 80.0f };
+s16 D_80ABD76C[] = { 0xC000, 0x4000 };
+CollisionHeader* D_80ABD770[] = { object_tokei_tobira_Colheader_0012B0, object_tokei_tobira_Colheader_001590 };
+f32 D_80ABD778[] = { 1.0f, -1.0f };
+Gfx* D_80ABD780[] = { object_tokei_tobira_DL_001108, object_tokei_tobira_DL_0013E8 };
 
-extern InitChainEntry D_80ABD750[];
+void ObjTokeiTobira_Init(Actor* thisx, PlayState* play) {
+    ObjTokeiTobira* this = THIS;
+    s32 pad;
+    s32 params = OBJTOKEITOBIRA_GET_TYPE(&this->dyna.actor);
+    Vec3f pos; // Object's position
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Init.s")
+    Actor_ProcessInitChain(&this->dyna.actor, D_80ABD750);
+    DynaPolyActor_Init(&this->dyna, DYNA_TRANSFORM_POS | DYNA_TRANSFORM_ROT_Y);
+    DynaPolyActor_LoadMesh(play, &this->dyna, D_80ABD770[params]);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Destroy.s")
+    if (params == 0) {
+        Actor_SpawnAsChild(&play->actorCtx, &this->dyna.actor, play, ACTOR_OBJ_TOKEI_TOBIRA,
+                           this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z,
+                           this->dyna.actor.shape.rot.x, this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z, 1);
+    }
```

Comment:
Please add a type enum to the header of this actor like this and a params macro next to the getter macro
```c
#define OBJTOKEITOBIRA_PARAM(type) (type)

typedef enum ObjTokeiTobiraType {
    /* 0 */ OBJTOKEITOBIRA_TYPE_0,
    /* 1 */ OBJTOKEITOBIRA_TYPE_1
} ObjTokeiTobiraType;
```
Then replace any comparison of the `type` to some value of this enum. And when invoking an spawn function then use the PARAM macro like this
```suggestion
    if (type == OBJTOKEITOBIRA_TYPE_0) {
        Actor_SpawnAsChild(&play->actorCtx, &this->dyna.actor, play, ACTOR_OBJ_TOKEI_TOBIRA,
                           this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z,
                           this->dyna.actor.shape.rot.x, this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z, OBJTOKEITOBIRA_PARAM(OBJTOKEITOBIRA_TYPE_1));
    }
```